src/dso_api/dynamic_api/remote/views.py

```diff
@@ -23,6 +28,55 @@
 logger = logging.getLogger(__name__)
 
 
+def _rewrite_links(data, old_prefix: str, new_prefix: str, in_links: bool = False):
+    """Rewrite hrefs in _links sections that start with old_prefix
+    to start with new_prefix instead.
+
+    May modify data destructively.
+    """
+    if isinstance(data, list):
+        for i in range(len(data)):
+            data[i] = _rewrite_links(data[i], old_prefix, new_prefix, in_links)
+            return data
+    elif isinstance(data, dict):
+        if in_links and data.get("href", "").startswith(old_prefix):
+            data["href"] = new_prefix + data["href"][len(old_prefix) :]
+            return data
+        return {
+            k: _rewrite_links(v, old_prefix, new_prefix, in_links or k == "_links")
+            for k, v in data.items()
+        }
+    else:
+        return data
+
+
+class HaalCentraalBAG(View):
+    """View that proxies Haal Centraal BAG.
+
+    This is simple pass-through proxy: we change the url, send the request on,
+    then fix up the response so that self/next/prev links point to us instead of HC.
+    """
+
+    _BASE_URL = settings.DATAPUNT_API_URL + "v1/haalcentraal/bag/"  # XXX use reverse?
+
+    def __init__(self):
+        super().__init__()
+        self.endpoint = "https://api.bag.acceptatie.kadaster.nl/esd/huidigebevragingen/v1/"
```

Comment:
Why not use the variable from settings for self.endpoint?

Reply:
Because I forgot to update this. Fixed now.